### PR TITLE
Record uniqueness validator

### PR DESCRIPTION
**Pull request by fefrain**

### CODE REVIEW

File: server/modules/record/manager/_recordManager/recordValidationManager.js [hunk 1]
```diff
@@ -59,9 +64,22 @@ const persistValidation = async (survey, record, nodesValidation, tx) => {
   await RecordRepository.updateValidation(surveyId, Record.getUuid(record), recordValidationUpdated, tx)
 }
 
+const validateOtherRecordsUniquenessAndPersistValidation = async (survey, record, excludeRecordFromCount, t) => {
```
Comment (minotogna) [inline server/modules/record/manager/_recordManager/recordValidationManager.js]: should we call this method validateRecordsUniquenessAndPersistValidation?

File: server/modules/record/manager/_recordManager/recordUpdateManager.js [hunk 2]
```diff
@@ -93,28 +93,57 @@ const deleteRecordsPreview = async (surveyId) =>
...
+      const { record: updatedRecord, updatedNodesAndDependents } = await _onNodesUpdate(
+        survey,
+        nodesUpdated,
+        nodesUpdateListener,
+        nodesValidationListener,
+        t,
+      )
+      await _afterNodesUpdate(user, survey, updatedRecord, updatedNodesAndDependents, t)
+
+      return updatedRecord
```
Comment (minotogna) [inline server/modules/record/manager/_recordManager/recordUpdateManager.js]: persistNode and deleteNode do the same thing, expect for the persist/delete node. should we have just a method with all the before and after update logic? e.g ```_updateNodeAndValidateKeys```?

File: server/modules/record/manager/_recordManager/recordUpdateManager.js [hunk 3]
```diff
@@ -93,28 +93,57 @@ const deleteRecordsPreview = async (surveyId) =>
  */
```
Comment (minotogna) [inline server/modules/record/manager/_recordManager/recordUpdateManager.js]: should the sequence diagram be updated as well?

File: server/modules/record/manager/_recordManager/recordValidationManager.js (hunk 1 above)
Comment (fefrain) [reply server/modules/record/manager/_recordManager/recordValidationManager.js]: ok, the shorter the better...

File: server/modules/record/manager/_recordManager/recordUpdateManager.js (hunk 2 above)
Comment (fefrain) [reply server/modules/record/manager/_recordManager/recordUpdateManager.js]: or something similiar, right

File: server/modules/record/manager/_recordManager/recordUpdateManager.js (hunk 3 above)
Comment (fefrain) [reply server/modules/record/manager/_recordManager/recordUpdateManager.js]: of course

File: server/modules/record/manager/_recordManager/recordUpdateManager.js (hunk 2 above)
Comment (minotogna) [reply server/modules/record/manager/_recordManager/recordUpdateManager.js]: why not to call the method for what it really does? the word perform doesn't really tell me what the method does.. don't you think so?

**Pull request merged**